client/electron/scripts/sign-latest-release.py

```diff
@@ -0,0 +1,137 @@
+        timestamp = artefact["updated_at"]
+        head = artefact["workflow_run"]["head_branch"]
+        if name != ARTIFACT_NAME:
+            continue
+        if not head.startswith("v"):
+            continue
+        if version is not None and head != version:
+            continue
+        return name, size, url, timestamp, head
+    if version is None:
+        raise ValueError(f"Artifact {ARTIFACT_NAME} not found")
+    else:
+        raise ValueError(f"Artifact {ARTIFACT_NAME} not found for version {version}")
+
+
+def download_artifact(url: str, token: str, size: int, destination: pathlib.Path) -> None:
+    data = get(url, token)
+    if destination.exists() and destination.stat().st_size == size:
+        print(f"Skipping download, file already exists: {destination}")
+        return
+    print(f"Downloading: {url}")
+    print(f"Into: {destination}")
+    with destination.open("wb") as f:
+        with tqdm.tqdm(total=size) as progress_bar:
+            for chunk in data.stream():
+                f.write(chunk)
+                progress_bar.update(len(chunk))
+
+
+def unzip_artifact(path: pathlib.Path) -> pathlib.Path:
+    assert path.suffix == ".zip"
+    destination = path.parent / path.stem
+    if destination.exists():
+        shutil.rmtree(destination)
+    print(f"Unzipping {path} to {destination}...")
+    with ZipFile(str(path), "r") as zip_ref:
+        zip_ref.extractall(destination)
+    return destination
+
+
+def check_version(path: pathlib.Path, version: str) -> None:
+    with (path / "package.json").open() as f:
+        package = json.load(f)
+    if f"v{package['version']}" != version:
+        raise ValueError(f"Version mismatch: {version} != v{package['version']}")
+    print(f"Version check passed on {path}")
+
+
+def sign_release(path: pathlib.Path) -> None:
+    script = (path / SIGN_SCRIPT).resolve()
+    subprocess.run([script], check=True, cwd=path)
+    upload_dir = (path / "upload").resolve()
+    subprocess.Popen(f'explorer "{upload_dir}"', shell=True)
+
+
+def main(version: str | None = None) -> None:
+    token = get_github_token()
+    name, size, url, timestamp, version = get_artifact(BASE_URL, token, version)
+    print(f"Found artifact: {name} version {version} updated at {timestamp}")
+    zip_path = pathlib.Path(f"{name}-{version}.zip")
+    download_artifact(url, token, size, zip_path)
+    destination = unzip_artifact(zip_path)
+    check_version(destination, version)
+    sign_release(destination)
+    print("Done!")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Sign the latest release artifact.")
+    parser.add_argument(
+        "--version",
+        type=str,
+        help="Specify the version of the artifact to sign (e.g., v1.2.3). If not provided, the latest version will be used.",
+    )
+    args = parser.parse_args()
+    version = args.version if args.version.startswith("v") else f"v{args.version}"
+    main(version)
```

Comment:
Sure about "If not provided, the latest version will be used"?

I didn't run the script, but it seems to me that if not provided, `version` is set to `"v"` on line 136, which would result in Artifact not found error.

Reply:
`args.version` would be `None` so that would be an attribute error, good catch

Reply:
b7a861c8fd56a84435a88315563c128a4797b220